pkg/vpcmodel/explainabilityPrint.go

```diff
@@ -478,28 +478,46 @@ func FilterKindName(filterLayer string) string {
 	}
 }
 
-// for a given filter layer (e.g. sg) returns a string of the allowing tables (note that denying tables are excluded),
-// and the name of the denying table, if any
+// for a given filter layer (e.g. sg) returns a string of the allowing tables,
+// and string of the denying table(s) if the connection is indeed denied
+// the connection is denied if the denying table is NACL (in which case there is only one attached table)
+// or if the table is SG and then there are no allowing tables. This is since one SG suffice to enable connection
 func pathFiltersSingleLayerStr(allRulesDetails *rulesDetails, filterLayerName string,
-	rules []RulesInTable) (allowPath, denyTable string) {
+	rules []RulesInTable) (allowPath, denyPath string) {
 	filtersToActionMap := allRulesDetails.listFilterWithAction(filterLayerName, rules)
-	strSlice := []string{}
+	allowTablesSlice := []string{}
+	denyTablesSlice := []string{}
+	// Is there at least one allowing SG? if so, we ignore non-allowing SGs (if any)
+	ignoreBlocking := false
+	if filterLayerName == SecurityGroupLayer {
+		for _, effect := range filtersToActionMap {
```

Comment:
`allows` might be a better name than `effect`

Reply:
done